commit 4d474c67e8b8c30adced330e848734123dbe7af6
Author: Yang Bo <yangbo@deepin.com>
Date:   Fri Nov 17 16:27:22 2017 +0800

    UBUNTU: SAUCE: (efi-lockdown) efi: Sanitize boot_params in efi stub
    
    The efi stub will set the value of boot_params.secure_boot
    without first checking whether boot_params has been sanitized. If
    they have not, the value of secure_boot will be cleared later
    when boot_params is sanitized. This currently happens with grub
    as it currently does not clear the sentinel, and thus the kernel
    cannot determine the secure boot state.
    
    Since the efi stub is modifying a field in an area subject to
    sanitization, it must first sanitize boot_params if needed. Later
    sanitization by the decompressor will do nothing as the sentinel
    value will have been cleared.
    
    Signed-off-by: Seth Forshee <seth.forshee@canonical.com>
    Signed-off-by: Yang Bo <yangbo@deepin.com>

diff --git a/arch/x86/boot/compressed/eboot.c b/arch/x86/boot/compressed/eboot.c
index c2213a74b..db195f519 100644
--- a/arch/x86/boot/compressed/eboot.c
+++ b/arch/x86/boot/compressed/eboot.c
@@ -14,6 +14,7 @@
 #include <asm/e820/types.h>
 #include <asm/setup.h>
 #include <asm/desc.h>
+#include <asm/bootparam_utils.h>
 
 #include "../string.h"
 #include "eboot.h"
@@ -1012,6 +1013,8 @@ struct boot_params *efi_main(struct efi_config *c,
 	 */
 	enable_reset_attack_mitigation();
 
+	sanitize_boot_params(boot_params);
+
 	/*
 	 * If the boot loader gave us a value for secure_boot then we use that,
 	 * otherwise we ask the BIOS.
